script.js

```diff
@@ -1,12 +1,156 @@
+    episodeNameNumTextHtml.textContent = episodeNameNum;
+
+    const imageContainerHtml = document.createElement("div");
+    imageContainerHtml.classList.add("episode-image-container");
+
+    // Get the episode image and description
+    const episodeImageHtml = document.createElement("img");
+    episodeImageHtml.classList.add("medium-img");
+    episodeImageHtml.src = episodeImageSrc;
+
+    const episodeDescriptionContainerHtml = document.createElement("div");
+    episodeDescriptionContainerHtml.classList.add(
+      "episode-description-container"
+    );
+
+    const episodeDescriptionTextHtml = document.createElement("p");
+    episodeDescriptionTextHtml.classList.add("episode-description-text");
+    episodeDescriptionTextHtml.innerHTML = episodeDescription;
+
+    // Update info about displayed episodes
+    searchResultDisplayHtml.textContent = `Displaying ${cardsContainerHtml.length}/${episodeList.length} episodes`;
+
+    // Add the episode card elements to the container
+    episodeNameNumHolderHtml.appendChild(episodeNameNumTextHtml);
+    episodeCardHtml.appendChild(episodeNameNumHolderHtml);
+    episodeCardHtml.appendChild(imageContainerHtml);
+    imageContainerHtml.appendChild(episodeImageHtml);
+    episodeDescriptionContainerHtml.appendChild(episodeDescriptionTextHtml);
+    episodeCardHtml.appendChild(episodeDescriptionContainerHtml);
+    cardsContainerHtml.appendChild(episodeCardHtml);
+    rootHtml.appendChild(cardsContainerHtml);
+  });
 }
 
-window.onload = setup;
+// level 200
+
+// function to search
+
+function displaySearchedEpisodes(inputValue) {
+  // creating a fresh array which only holds the episodes that match our search criteria
+  const filteredEpisodes = allEpisodes.filter((episode) => {
+    return (
+      episode.name.toLowerCase().includes(inputValue) ||
+      episode.summary.toLowerCase().includes(inputValue)
+    );
+  });
+
+  // Display searchedEpisodes
+  if (filteredEpisodes.length !== 0) {
+    rootHtml.innerHTML = "";
+    makePageForEpisodes(filteredEpisodes);
+  }
+  searchResultDisplayHtml.textContent = `Displaying ${filteredEpisodes.length}/${allEpisodes.length} episodes`;
+}
+
+// List of event listeners (Should be always at the bottom)
+
+searchInputHtml.addEventListener("input", (event) => {
+  const inputValue = event.target.value.trim().toLowerCase();
+  displaySearchedEpisodes(inputValue);
+});
+
+// level 300
+
+// jump to episode function
+
+function jumpToEpisode(event) {
+  // console.log(event.target);
+  const episodeSelectHTML = document.querySelector("#select-html"); // this is the dropdown
+  const position = episodeSelectHTML.value; // we set a variable for the value which is the number
+  // displaySearchedEpisodes(episodeSelectHTML);
+  const episodeSelectHTMLId = "episode-card" + position; // position is from the dropdown list; we are making the id will will find
+  document.getElementById(episodeSelectHTMLId).scrollIntoView({
```

Comment:
Good scroll to view, but there are problems. When we displayed filtered episodes and then select the episode, we're getting mistakes. Since there are several ways to display them, I recommend to display selected episode vie filter, this way we will avoid this issue

Reply:
You are right. I will need to fix that too.